hydroflow_lang/src/graph/ops/halfjoin.rs

```diff
@@ -0,0 +1,143 @@
+use super::{
+    parse_single_lifetime, OperatorConstraints, OperatorWriteOutput, Persistence, WriteContextArgs,
+    WriteIteratorArgs, RANGE_1,
+};
+
+use quote::quote_spanned;
+use syn::parse_quote;
+
+/// > 2 input streams of type <(K, V1)> and <(K, V2)>, 1 output stream of type <(K, (V1, V2))>
```

Comment:
fix comment? will halfjoin be exposed at user level? why is it `pub const JOIN`?

Reply:
Removed, join will have separate left and right persistences